src/main.c

```diff
@@ -448,7 +453,12 @@ np2srv_diff_check_cb(sr_session_ctx_t *session, const struct lyd_node *diff)
     }
 
     sr_session_get_orig_data(session, 1, NULL, (const void **)&user);
-    if ((node = ncac_check_diff(diff, user))) {
+    err_info = sr_nacm_check_diff(diff, user, &node);
```

Comment:
We have talked about removing this whole callback, is that still the plan?

Reply:
I think it is, but I'm not 100% sure I'll be able to do that by myself. It seems that the only place it is called from is `sr_changes_notify_store`, which appears to be some sort of a subscription-related function. I would think that the diff check would be called in something like `sr_apply_changes`. Sorry, my understanding of sysrepo code is minimal... :(

Reply:
That is fine but my first idea, which I have mentioned before, would be to basically just set a flag in the NACM variant of `sr_apply_changes()` and pass it until the point where the callback is called and there just run the code currently in the *netopeer2* diff check callback in case the flag is set, nothing else.

Reply:
Actually, the user should be a parameter of `sr_nacm_apply_changes()` or whatever the name and also serve as a flag whether to apply NACM or not. All the other NACM functions have this argument so it only makes sense.